### PR TITLE
Call API to restart all services on node reboot

### PR DESCRIPTION
**Pull request by vatsa287**
With [PR](https://github.com/kadalu/moana/pull/160/commits/2e7109caf92e20271f97b495722247ff6082659c) to rerouting url request on agent to mgr being implemented. We can avoid to check for agent or mgr services. Since anyway it will be forwarded.

Signed-off-by: Shree Vatsa N <vatsa@kadalu.io>

### CODE REVIEW

File: mgr/src/server/server.cr [hunk 1]
```diff
@@ -116,18 +116,13 @@ module StorageMgr
     if GlobalConfig.agent
```
Comment (aravindavk) [inline mgr/src/server/server.cr]: Remove if condition. It is required for both agent and Mgr

File: mgr/src/server/plugins/node_add_remove.cr [hunk 2]
```diff
@@ -81,8 +81,9 @@ node_action ACTION_NODE_REMOVE do |data, _env|
   # TODO: Handle error while writing node data
   File.write(data_file, local_node_data.to_json)
 
-  # Remove Agent/Manager identifier
-  Datastore.remove_manager_agent
+  # Remove Agent identifier
+  # Leave it as it is if it is Manager
+  Datastore.remove_agent
```
Comment (aravindavk) [inline mgr/src/server/plugins/node_add_remove.cr]: why this change is needed. Don't leave Mgr or Agent identifier on node remove.

File: mgr/src/server/server.cr (hunk 1 above)
Comment (aravindavk) [reply mgr/src/server/server.cr]: Sorry for the confusion. We also need else block as existing.

File: mgr/src/server/server.cr [hunk 3]
```diff
@@ -113,21 +113,25 @@ module StorageMgr
...
-    else
-      services = Datastore.list_services(
-        GlobalConfig.local_node.pool_name,
-        GlobalConfig.local_node.name
-      )
+      loop do
+        begin
+          # TODO: Handle Authentication
+          # TODO: Get Server URL from GlobalConfig.local_node
+          url = "http://localhost:#{Kemal.config.port}/api/v1/pools/#{GlobalConfig.local_node.pool_name}/nodes/#{GlobalConfig.local_node.name}/services"
```
Comment (aravindavk) [inline mgr/src/server/server.cr]: Change the URL to use mgr url. Construct the URL like this https://github.com/kadalu/moana/blob/main/mgr/src/server/routes.cr#L35-L41

Comment (aravindavk) [reply mgr/src/server/server.cr]: Proxy will not work here as discussed previously since the local server is not started yet. So use the Mgr URL as above comment

File: mgr/src/server/server.cr [hunk 4]
```diff
@@ -113,21 +113,25 @@ module StorageMgr
...
+          resp = HTTP::Client.get(url)
+          # # TODO: Exit on error
+          if resp.status_code == 200
+            services = Array(MoanaTypes::ServiceUnit).from_json(resp.body)
+            break
+          end
+          Log.debug &.emit("Fetch services of this node")
```
Comment (aravindavk) [inline mgr/src/server/server.cr]: Failed to fetch? Also add `resp.status_code` to the log

File: mgr/src/server/plugins/node_add_remove.cr (hunk 2 above)
Comment (vatsa287) [reply mgr/src/server/plugins/node_add_remove.cr]: No Added the previous commit by mistake, will revert.

File: mgr/src/server/server.cr (hunk 4 above)
Comment (vatsa287) [reply mgr/src/server/server.cr]: Done

File: mgr/src/server/server.cr (hunk 1 above)
Comment (vatsa287) [reply mgr/src/server/server.cr]: Retained else block.

**Review by aravindavk — APPROVED** [review]
